### PR TITLE
Create scaffold and logic controller movements 

### PR DESCRIPTION
**Pull request by Nany-fork**
For not authorized merged [PR](https://github.com/Nany-fork/ledger_account/pull/7).

### CODE REVIEW

File: app/models/movement.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class Movement < ApplicationRecord
+  belongs_to :movement_type
+  belongs_to :account
```
Comment (luisAzcuaga) [inline app/models/movement.rb:3]: Thanks to this relation, you could easily do what I told in the previous comment. If you don't want to reach me, then look for the `before_create` hook. :)

File: app/views/movement_types/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      <ul>
+        <% movement_type.errors.each do |error| %>
+          <li><%= error.full_message %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
+
+  <div>
+    <%= form.label :name, style: "display: block" %>
```
Comment (luisAzcuaga) [inline app/views/movement_types/_form.html.erb:15]: Avoid using inline styles, always prefer using class! ✨  (code looks cleaner this way)

File: app/views/movements/_movement.html.erb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<div id="<%= dom_id movement %>">
+  <tbody>
+    <tr>
+      <td><%= movement.name %></td>
```
Comment (luisAzcuaga) [inline app/views/movements/_movement.html.erb:4]: I always love to recommend using tags, helper. (Is up to you changing it)
This line would change to:
```erb
<%= tag.td movement.name %>
```
That doesn't save up too much space, huh?
But take a look at this scenario
```erb
<% if @something.present? %>
  <td><%= @something.name %></td>
<% end %>
```
Could be changed to:
```erb
<%= tag.td @something.name if @something.present? %>
```

**Review by luisAzcuaga — CHANGES_REQUESTED** [review]
Nice work, just take a look at my suggestions! ✨

File: app/controllers/movements_controller.rb [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    @movement = Movement.new
+  end
+
+
+  def edit
+  end
+
+ 
+  def create
+    # if params[:movement][:movement_type_id] == '1'
```
Comment (itsmeurbi) [inline app/controllers/movements_controller.rb:26]: ?

File: app/views/movement_types/_form.html.erb (hunk 2 above)
Comment (itsmeurbi) [reply app/views/movement_types/_form.html.erb:15]: @Nany-fork

**Review by itsmeurbi — CHANGES_REQUESTED** [review]
Didn't you see my comments [here](https://github.com/Nany-fork/ledger_account/pull/7/files)?
There still a couple of things pending, like all model validations

We had a conversation yesterday about how to create your initial categories and movement types. You need to populate that values through the seeds file. There's no need for controllers if nobody will manage those resources 🤔

**Review by itsmeurbi — APPROVED** [review]
I'd say looks good to go 👍

**Comment by Nany-fork** [comment]
## Creation of movements and categories 

# validations movement 
- minumum of the name and descripcion
- you can not register a date later than the current one
- you can register a new category_movement
-update the balance with (incomes, egress, payment)

**Pull request merged**